plugin.php

```diff
@@ -584,12 +588,24 @@ function rest_handle_options_request( $response, $handler, $request ) {
 
 		foreach ( $endpoints as $endpoint ) {
 			$accept = array_merge( $accept, $endpoint['methods'] );
+
+			$methods = array_keys( $endpoint['methods'] );
+			$method = array_shift( $methods );
+
+			if ( empty( $body['request_args'][ $method ] ) ) {
+				$body['request_args'][ $method ] = $endpoint['args'];
+			}
+
+			if ( empty( $body['schema'] ) && ! empty( $endpoint['schema'] ) ) {
+				$body['schema'] = $endpoint['schema'];
+			}
 		}
 		break;
 	}
 	$accept = array_keys( $accept );
 
 	$response->header( 'Accept', implode( ', ', $accept ) );
+	$response->set_data( (object) $body );
```

Comment:
Why is this cast to an object? Pretty much everything else return an array / `ArrayAccess`